src/perf_tool_backend.ml

```diff
@@ -106,43 +106,65 @@ module Perf_line = struct
   let report_itraces = "b"
   let report_fields = "pid,tid,time,flags,addr,sym,symoff"
 
-  let to_event line =
+  let saturating_sub_i64 a b =
+    match Int64.(to_int (a - b)) with
+    | None -> Int.max_value
+    | Some offset -> offset
+  ;;
+
+  let parse_kind_exn kind : Backend_intf.Event.kind list =
+    match String.strip kind with
+    | "call" -> [ Call ]
+    | "return" -> [ Return ]
+    | "tr strt" -> [ Start ]
+    | "tr end" -> [ End None ]
+    | "tr end  call" -> [ End Call ]
+    | "tr end  return" -> [ End Return ]
+    | "tr end  syscall" -> [ End Syscall ]
+    | "jmp" -> [ Jump ]
+    | "jcc" -> [ Jump ]
+    (* CR cgaebel: I have no idea if these are right but they led to the least broken traces out
+       of the options I tried. I didn't try many. *)
+    | "syscall" -> [ Call ]
+    | "int" -> [ Call ]
```

Comment:
Played around with this a bit. I'm not sure what V8 is using interrupts for, but modelling them as jumps rather than calls makes the traces look a lot better:

![image](https://user-images.githubusercontent.com/1403503/160533936-3e8d8620-4e22-4d8d-9b45-9a515c705ee3.png)

Reply:
I'm going to add software interrupts in a separate feature.